### PR TITLE
Add most_recent_feature_version and tip_version

### PR DESCRIPTION
**Pull request by maxandersen**

### CODE REVIEW

**Comment by maxandersen** [comment]
fixes #6

File: net.adoptopenjdk.v3.vanilla/src/main/java/net/adoptopenjdk/v3/vanilla/internal/AOV3AST.java [hunk 1]
```diff
@@ -61,6 +61,13 @@
 
     @JsonProperty(value = "most_recent_lts", required = true)
     BigInteger mostRecentLTS = BigInteger.ZERO;
+
+    @JsonProperty(value = "most_recent_feature_version", required = true)
+    BigInteger getMostRecentFeatureVersion;
+
+    @JsonProperty(value = "tip_version", required = true)
+    BigInteger tipVersion;
```
Comment (gastaldi) [inline net.adoptopenjdk.v3.vanilla/src/main/java/net/adoptopenjdk/v3/vanilla/internal/AOV3AST.java:69]:
```suggestion
    BigInteger tipVersion = BigInteger.ZERO;
```

File: net.adoptopenjdk.v3.vanilla/src/main/java/net/adoptopenjdk/v3/vanilla/internal/AOV3AST.java [hunk 2]
```diff
@@ -61,6 +61,13 @@
 
     @JsonProperty(value = "most_recent_lts", required = true)
     BigInteger mostRecentLTS = BigInteger.ZERO;
+
+    @JsonProperty(value = "most_recent_feature_version", required = true)
+    BigInteger getMostRecentFeatureVersion;
```
Comment (gastaldi) [inline net.adoptopenjdk.v3.vanilla/src/main/java/net/adoptopenjdk/v3/vanilla/internal/AOV3AST.java:66]:
```suggestion
    BigInteger getMostRecentFeatureVersion = BigInteger.ZERO;
```

Comment (quintesse) [reply net.adoptopenjdk.v3.vanilla/src/main/java/net/adoptopenjdk/v3/vanilla/internal/AOV3AST.java:66]: No need for that if the value is required, right?

Comment (io7m) [reply net.adoptopenjdk.v3.vanilla/src/main/java/net/adoptopenjdk/v3/vanilla/internal/AOV3AST.java:66]: I believe that the current Jackson semantics are such that, if you specify a default value, it effectively turns off the check that requires the value to be present in the JSON.

**Comment by io7m** [comment]
@maxandersen Could you add yourself as a contributor to the POM?

See: http://maven.apache.org/pom.html#contributors

I'm happy to merge this, it looks fine to me.

**Comment by maxandersen** [comment]
is there a pom.xml that has some already ?

**Comment by io7m** [comment]
There isn't, but the documentation I linked to has a good example:

```
<contributors>
    <contributor>
      <name>Noelle</name>
      <email>some.name@gmail.com</email>
      <url>http://noellemarie.com</url>
      ...
    </contributor>
  </contributors>
```

I think just the `name` and `url` fields would be enough. You don't have to make your email address public. :smile:

**Comment by joschi** [comment]
I'm wondering if the Jackson object mapper shouldn't be configured to be more lenient and ignore unknown properties for this project: https://github.com/FasterXML/jackson-docs/wiki/JacksonHowToIgnoreUnknown

Otherwise, the client will break again if a new property is added to the AdoptOpenJDK API.

Or put differently: ["Be conservative in what you send, be liberal in what you accept."](https://en.wikipedia.org/wiki/Robustness_principle)

**Comment by io7m** [comment]
I actually strongly disagree with Postel's law in general, and this RFC states my objections to it better than I would:

https://tools.ietf.org/html/draft-iab-protocol-maintenance-04

However, I do agree in this case that it would be better if the API ignored additional unrecognized fields. I want to ensure that the API treats missing required fields as an error, but doesn't care about extra fields it sees.

**Comment by maxandersen** [comment]
ack, done.

**Comment by io7m** [comment]
Thanks very much!

I should be able to work on adjusting the defaults so that extra fields are ignored at some point this week.

**Pull request merged**